pixelgl/gltriangles.go

```diff
@@ -48,7 +48,7 @@ func (gt *GLTriangles) VertexSlice() *glhf.VertexSlice {
 }
 
 // Shader returns the GLTriangles's associated shader.
-func (gt *GLTriangles) Shader() *glhf.Shader {
+func (gt *GLTriangles) Shader() *GLShader {
```

Comment:
This looks like a breaking change. Will this break user's code?

Reply:
Yeah that's a downside to this update, that line and the constructor for the GLTriangles are a breaking change.

Although, I *think* it should be fairly safe given that the GLTriangles shader doesn't seem to be used anywhere else in Pixel. So the only time this would be breaking is if someone is creating GLTriangles directly with a glhf shader and using that shader (they would have to be doing shader stuff directly since the Canvas only uses its own shader); everyone else should be fine.

Completely understand if we don't want to take that risk.

Reply:
Will this breaking change be mitigated with Go modules? @delp

Reply:
Do you mean that modules makes it easier to revert to an older version tag that doesn't have this change? Yes, I think that would be true.

Also since we are not in 1.0 we are not guaranteeing that version increments don't have breaking changes.